### PR TITLE
feat(x/ecocredit): default values for new ORM params

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #1158

This pull request updates default genesis values to include a default credit type and a default allowed denom to make it easier for starting a local testnet and then testing against that local testnet.

In order to support a default allowed denom (a marketplace type), a separate `genesis` package was created in the root of the module directory (previously in `core` - was probably not the right place anyways). Import cycles were violated in `module` and in the root of the ecocredit module so a separate `genesis` package was the simplest solution.

Validation functions for class, class issuers, project, and batch were moved to respective type files for each within `core`.

Manual tests: https://hackmd.io/WYHeLuIQSKOnLlJVGsFVnw (including upgrade tests just to be sure this does not affect the upgrade although these changes only affect starting a new chain without a genesis file as they should).

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/core/types_batch.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package core
```
Comment (ryanchristo) [inline x/ecocredit/core/types_batch.go:1]: No changes here. Moved from `core/genesis.go` (and what is now `genesis/genesis.go`).

File: x/ecocredit/core/types_class.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package core
```
Comment (ryanchristo) [inline x/ecocredit/core/types_class.go:1]: No changes here. Moved from `core/genesis.go` (and what is now `genesis/genesis.go`).

File: x/ecocredit/core/types_class_issuer.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package core
```
Comment (ryanchristo) [inline x/ecocredit/core/types_class_issuer.go:1]: No changes here. Moved from `core/genesis.go` (and what is now `genesis/genesis.go`).

File: x/ecocredit/core/types_project.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package core
```
Comment (ryanchristo) [inline x/ecocredit/core/types_project.go:1]: No changes here. Moved from `core/genesis.go` (and what is now `genesis/genesis.go`).

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**